src/Datasource/ModelAwareTrait.php

```diff
@@ -119,15 +118,28 @@ public function loadModel($modelClass = null, $modelType = null)
     }
 
     /**
-     * Register a callable to generate repositories of a given type.
+     * Override a existing callable to generate repositories of a given type.
      *
      * @param string $type The name of the repository type the factory function is for.
      * @param callable $factory The factory function used to create instances.
      * @return void
+     * @deprecated 3.3.0 Use either ModelFactory::add or ModelAwareTrait::overrideModelFactory instead.
      */
     public function modelFactory($type, callable $factory)
     {
-        $this->_modelFactories[$type] = $factory;
+        $this->overrideModelFactory($type, $factory);
+    }
+
+    /**
+     * Override a existing callable to generate repositories of a given type.
+     *
+     * @param string $type The name of the repository type the factory function is for.
+     * @param callable $factory The factory function used to create instances.
+     * @return void
+     */
+    public function overrideModelFactory($type, callable $factory)
```

Comment:
I don't like the name `overrideModelFactory()`. Instead of deprecating `modelFactory()` can't it be just modified to do what `overrideModelFactory()` does?